discovery-provider/src/tasks/index.py

```diff
@@ -125,36 +126,58 @@ def initialize_blocks_table_if_necessary(db: SessionManager):
     return target_blockhash
 
 
-def get_latest_block(db: SessionManager):
-    latest_block = None
```

Comment:
moved the return for this inside the with. previously was declaring this outside with, setting inside with, returning outside with